addOns/client/src/main/java/org/zaproxy/addon/client/ui/PopupMenuExportClientMap.java

```diff
@@ -0,0 +1,290 @@
+            }
+            fw.write(key);
+            fw.write(": ");
+            ObjectMapper mapper =
+                    new ObjectMapper(
+                            new YAMLFactory()
+                                    .disable(Feature.WRITE_DOC_START_MARKER)
+                                    .enable(Feature.LITERAL_BLOCK_STYLE));
+            fw.write(mapper.writeValueAsString(value));
+        }
+
+        private static boolean outputKVIfNotEmpty(
+                BufferedWriter fw, String indent, boolean first, String key, Object value)
+                throws IOException {
+            if (value == null
+                    || (value instanceof String strVal && strVal.isEmpty())
+                    || (value instanceof Integer num && num.intValue() <= 0)) {
+                return first;
+            }
+            indent = indent + "  ";
+            if (value instanceof String strVal) {
+                outputKV(fw, indent, first, key, strVal);
+                return false;
+            } else {
+                outputKV(fw, indent, first, key, value);
+                return false;
+            }
+        }
+
+        private static void outputNode(BufferedWriter fw, ClientNode node, int level)
+                throws IOException {
+            // We could create a set of data structures and use jackson, but the format is very
+            // simple and this is much more memory efficient - it still uses jackson for value
+            // output
+            String indent = " ".repeat(level * 2);
+
+            outputKV(fw, indent, true, NODE_KEY, level == 0 ? ROOT_NODE_NAME : node.toString());
+
+            outputKV(fw, indent, false, NAME_KEY, node.getUserObject().getName());
+            outputKV(fw, indent, false, URL_KEY, node.getUserObject().getUrl());
+            outputKV(fw, indent, false, STORAGE_KEY, node.getUserObject().isStorage());
+            outputKV(fw, indent, false, VISITED_KEY, node.getUserObject().isVisited());
+            if (node.getUserObject().getComponents() != null
+                    && !node.getUserObject().getComponents().isEmpty()) {
+                fw.write(indent + "  ");
+                fw.write("components:\n");
+
+                boolean first = true;
+                for (ClientSideComponent component : node.getUserObject().getComponents()) {
+                    first = outputKVIfNotEmpty(fw, indent, first, "href", component.getHref());
+                    first = outputKVIfNotEmpty(fw, indent, first, "id", component.getId());
+                    first =
+                            outputKVIfNotEmpty(
+                                    fw, indent, first, "tagName", component.getTagName());
+                    first =
+                            outputKVIfNotEmpty(
+                                    fw, indent, first, "tagType", component.getTagType());
+                    first = outputKVIfNotEmpty(fw, indent, first, "text", component.getText());
```

Comment:
Text should not be included where it can be expected to be different each time, for example cookies and other storage components.